platform/darwin/src/MGLShapeSource.h

```diff
@@ -111,7 +45,7 @@ extern MGL_EXPORT const MGLShapeSourceOption MGLShapeSourceOptionSimplificationT
  ```
  */
 MGL_EXPORT
-@interface MGLShapeSource : MGLSource
+@interface MGLShapeSource : MGLAbstractShapeSource
```

Comment:
This PR is scheduled to land in iOS SDK v4.0.0, which allows backwards-incompatible changes. That means we can revisit the taxonomy proposed in https://github.com/mapbox/mapbox-gl-native/pull/6940#issuecomment-266232958. Specifically, we have the opportunity to choose a name for MGLAbstractShapeSource that doesn’t contain the word “abstract” and rename the `MGLShapeSourceOption`s to align with whatever class they’re used with. Any ideas?

Reply:
How about a single `MGLShapeSource` with a data source that provides either tiled or non-tiled data. It would combine the existing `MGLShapeSource` and `MGLProgrammaticShapeSource`.
I don't have a good idea of what the peer Source class for a GeoJSON source would be in this scenario - maybe it makes room for an actual `MGLGeoJSONSource`, but I suspect that was removed earlier for some very valid argument.

Reply:
I like this idea – would we combine `GeoJSONSource` and `CustomSource` into a single class to facilitate this simplification?

Reply:
That may be a future state, where `GeoJSONSource` is rebuilt on top of `CustomVectorSource`, but its not in the current plan

Reply:
I suppose that means MGLShapeSource would need to hold a `variant<GeoJSONSource, CustomSource>`? More reason to rename `CustomSource` to something more descriptive…

Reply:
Landed on using `CustomGeometrySource` in core and android to match other uses of _custom_.